events/notifications.go

```diff
@@ -1,71 +1,289 @@
-	var notifications []api.Notification
-	if err := tx.WithContext(ctx).Where("team_id = ? AND deleted_at IS NULL", responder.TeamID).Find(&notifications).Error; err != nil {
+func publishNotification(ctx *api.Context, event api.Event) error {
+	var (
+		connection = event.Properties["internal.connection"]
+		url        = event.Properties["internal.url"]
+		message    = event.Properties["internal.message"]
+	)
+
+	delete(event.Properties, "internal.url")
+	delete(event.Properties, "internal.message")
+	delete(event.Properties, "internal.connection")
+
+	return notification.Publish(ctx, connection, url, message, event.Properties)
+}
+
+// NotificationTemplate holds in data for notification
+// that'll be used by struct templater.
+type NotificationTemplate struct {
+	Message    string            `template:"true"`
+	Properties map[string]string `template:"true"`
+}
+
+func addNotificationEvent(ctx *api.Context, event api.Event) error {
+	notifications, err := notification.GetNotifications(ctx, event.Name)
+	if err != nil {
 		return err
 	}
 
+	if len(notifications) == 0 {
+		return nil
+	}
+
+	celEnv, err := getEnvForEvent(ctx, event.Name, event.Properties)
+	if err != nil {
+		return err
+	}
+
+	templater := template.StructTemplater{
+		Values:         celEnv,
+		ValueFunctions: true,
+		DelimSets: []template.Delims{
+			{Left: "{{", Right: "}}"},
+			{Left: "$(", Right: ")"},
+		},
+	}
+
 	for _, n := range notifications {
-		if valid, err := notification.IsValid(n.Config.Filter, responder); err != nil {
+		if !n.HasRecipients() || n.Template == "" {
+			continue
+		}
+
+		data := NotificationTemplate{
+			Message:    n.Template,
+			Properties: n.Properties,
+		}
+
+		if err := templater.Walk(&data); err != nil {
+			return fmt.Errorf("error templating notification: %w", err)
+		}
+
+		if valid, err := utils.EvalExpression(n.Filter, celEnv); err != nil {
 			return err
 		} else if !valid {
 			continue
 		}
 
-		event := api.Event{
-			ID:   uuid.New(),
-			Name: EventNotification,
-			Properties: map[string]string{
-				"notification_id": n.ID.String(),
-				"id":              responder.ID.String(),
-			},
+		if n.PersonID != nil {
+			var email string
+			if err := ctx.DB().Model(&models.Person{}).Select("email").Where("id = ?", n.PersonID).Find(&email).Error; err != nil {
+				return fmt.Errorf("failed to get email of person(id=%s); %v", n.PersonID, err)
+			} else {
+				newEvent := api.Event{
+					ID:   uuid.New(),
+					Name: EventNotificationPublish,
+					Properties: map[string]string{
+						"internal.message": data.Message,
+						"internal.url": fmt.Sprintf("smtp://%s:%s@%s:%s/?auth=Plain&FromAddress=%s&ToAddresses=%s",
+							url.QueryEscape(os.Getenv("SMTP_USER")),
+							url.QueryEscape(os.Getenv("SMTP_PASSWORD")),
+							os.Getenv("SMTP_HOST"),
+							os.Getenv("SMTP_PORT"),
+							url.QueryEscape(os.Getenv("SMTP_USER")),
+							url.QueryEscape(email),
```

Comment:
@moshloop, do you think it's alright to have credentials in the event queue? If not then we can encrypt it.

Notification config in team spec or custom notifications can also have credentials in the URL (if they do not use connection name).